metrics/src/main/java/org/aerogear/mobile/metrics/MetricsService.java

```diff
@@ -83,19 +70,17 @@ protected JSONObject metricsData(final Context context) throws JSONException {
      */
     public final void init(final Context context) {
         try {
-            final JSONObject data = metricsData(context);
+            JSONObject metrics = new JSONObject();
+            aggregateAllProviders(metrics, context);
 
             // Send request to backend
             HttpRequest request = httpService.newRequest();
-            request.post(metricsUrl, data.toString().getBytes());
+            request.post(metricsUrl, metrics.toString().getBytes());
             HttpResponse response = request.execute();
 
             // Async response handling
-            response.onComplete(() -> {
-                if (response.getStatus() != 200) {
-                    logger.error(TAG, "Error sending metrics data");
-                }
-            });
+            response.onComplete(new MetricsResponseHandler(response));
+            response.waitForCompletionAndClose();
```

Comment:
This will block the main thread which means that init should be called from a background thread.  Is that the assumption?

Reply:
No, better to remove that line and just let it do it's thing in the background.